dlt/common/schema/schema.py

```diff
@@ -177,6 +184,71 @@ def coerce_row(self, table_name: str, parent_table: str, row: StrAny) -> Tuple[D
 
         return new_row, updated_table_partial
 
+    def resolve_evolution_settings_for_table(self, parent_table: str, table_name: str, schema_evolution_settings_override: TSchemaEvolutionSettings) -> TSchemaEvolutionModes:
+
+        def resolve_single(settings: TSchemaEvolutionSettings) -> TSchemaEvolutionModes:
+            settings = settings or {}
+            if isinstance(settings, str):
+                return TSchemaEvolutionModes(table=settings, column=settings, column_variant=settings)
+            return settings
+
+        # find table settings
+        table_with_settings = parent_table or table_name
+
+        # modes
+        table_evolution_modes = resolve_single(self.tables.get(table_with_settings, {}).get("schema_evolution_settings", {}))
+        schema_evolution_modes = resolve_single(self._settings.get("schema_evolution_settings", {}))
+        overide_modes = resolve_single(schema_evolution_settings_override)
+
+        # resolve to correct settings dict
+        settings = cast(TSchemaEvolutionModes, {**DEFAULT_SCHEMA_EVOLUTION_MODES, **schema_evolution_modes, **table_evolution_modes, **overide_modes})
+
+        return settings
+
+
+    def check_schema_update(self, parent_table: str, table_name: str, row: DictStrAny, partial_table: TPartialTableSchema, schema_evolution_settings_override: TSchemaEvolutionSettings) -> Tuple[DictStrAny, TPartialTableSchema]:
+        """Checks if schema update mode allows for the requested changes, filter row or reject update, depending on the mode"""
+
+        assert partial_table
+
+        # for now we defined the schema as new if there are no data columns defined
+        has_columns = self.has_data_columns
```

Comment:
I do not think we need this. if we have an empty schema that is completely frozen so be it- it will not change at all